src/nft/components/card/media.tsx

```diff
@@ -0,0 +1,329 @@
+  $hidden?: boolean
+}>`
+  width: 100%;
+  aspect-ratio: ${({ $aspectRatio }) => $aspectRatio};
+  transition: ${({ theme }) => `${theme.transition.duration.medium} ${theme.transition.timing.ease} transform`};
+  will-change: transform;
+  object-fit: contain;
+  visibility: ${({ $hidden }) => ($hidden ? 'hidden' : 'visible')};
+  background: ${({ theme, imageLoading }) =>
+    imageLoading && `linear-gradient(270deg, ${theme.backgroundOutline} 0%, ${theme.backgroundSurface} 100%)`};
+`
+
+export const NftImage = ({
+  src,
+  uniformAspectRatio = UniformAspectRatios.square,
+  setUniformAspectRatio,
+  renderedHeight,
+  setRenderedHeight,
+}: ImageProps) => {
+  const [noContent, setNoContent] = useState(!src)
+  const [loaded, setLoaded] = useState(false)
+
+  if (noContent) {
+    return <NoContentContainer height={getHeightFromAspectRatio(uniformAspectRatio, renderedHeight)} />
+  }
+
+  return (
+    <StyledMediaContainer>
+      <StyledImage
+        src={src}
+        $aspectRatio={getMediaAspectRatio(uniformAspectRatio, setUniformAspectRatio)}
+        imageLoading={!loaded}
+        draggable={false}
+        onError={() => setNoContent(true)}
+        onLoad={(e) => {
+          handleUniformAspectRatio(uniformAspectRatio, e, setUniformAspectRatio, renderedHeight, setRenderedHeight)
+          setLoaded(true)
+        }}
+      />
+    </StyledMediaContainer>
+  )
+}
+
+interface MediaProps {
+  mediaSrc?: string
+  tokenId?: string
+  shouldPlay: boolean
+  setCurrentTokenPlayingMedia: (tokenId: string | undefined) => void
+}
+
+const PlaybackButton = styled.div<{ pauseButton?: boolean }>`
+  display: ${({ pauseButton }) => (pauseButton ? 'block' : 'none')};
+  color: ${({ theme }) => theme.accentAction};
+  position: absolute;
+  height: 40px;
+  width: 40px;
+  z-index: 1;
+  margin-left: calc(100% - 50px);
+  transform: translateY(-76px);
+
+  @media screen and (max-width: ${BREAKPOINTS.sm}px) {
+    display: block;
+  }
+
+  ${StyledImageContainer}:hover & {
+    display: block;
+  }
+`
+
+const StyledVideo = styled.video<{
+  $aspectRatio?: string
+}>`
+  width: 100%;
+  aspect-ratio: ${({ $aspectRatio }) => $aspectRatio};
+`
+
+const StyledInnerMediaContainer = styled(Row)`
+  position: absolute;
+  left: 0px;
+  top: 0px;
+`
+
+export const NftVideo = ({
+  src,
+  mediaSrc,
+  tokenId,
+  uniformAspectRatio = UniformAspectRatios.square,
+  setUniformAspectRatio,
+  renderedHeight,
+  setRenderedHeight,
+  shouldPlay,
+  setCurrentTokenPlayingMedia,
+}: MediaProps & ImageProps) => {
+  const vidRef = useRef<HTMLVideoElement>(null)
+  const [noContent, setNoContent] = useState(!src)
+  const [imageLoaded, setImageLoaded] = useState(false)
+
+  useEffect(() => {
+    if (shouldPlay && vidRef.current) {
+      vidRef.current.play()
+    } else if (!shouldPlay && vidRef.current) {
+      vidRef.current.pause()
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [shouldPlay, vidRef.current])
```

Comment:
I did some testing and updated the dependencies to remove ignore and had no issues